### PR TITLE
MM-17468 - Improving performance of fetching threads

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Introduce a `fetchThreads` parameter to determine if the full thread is fetched with post queries, or just the post + root (to update the reply count / show the `Commented on...` prefix.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-17468

#### Related Pull Requests
https://github.com/mattermost/mattermost-server/pull/11980
https://github.com/mattermost/mattermost-webapp/pull/3549

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the [JavaScript driver](https://github.com/mattermost/mattermost-redux/blob/master/src/client/client4.js)

#### Test Information
This PR was tested on: [Device name(s), OS version(s)] 
Ubuntu 19.04

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "remote-redux-devtools": "0.5.16",
     "uglifyjs-webpack-plugin": "2.1.3",
     "webpack": "4.36.1",
+    "webpack-cli": "^3.3.6",
```
Comment (jespino) [inline package.json]: unrelated changes in this file

File: package-lock.json [hunk 2]
```diff
@@ -129,7 +129,7 @@
         },
         "minimist": {
           "version": "1.2.0",
-          "resolved": "https://registry.npmjs.org/minimist/-/minimist-1.2.0.tgz",
+          "resolved": "http://registry.npmjs.org/minimist/-/minimist-1.2.0.tgz",
```
Comment (jespino) [inline package-lock.json]: unrelated changes in this file

File: src/actions/posts.js [hunk 3]
```diff
@@ -623,12 +623,12 @@ export function getPosts(channelId, page = 0, perPage = Posts.POST_CHUNK_SIZE) {
...
 }
 
-export function getPostsUnread(channelId) {
+export function getPostsUnread(channelId, fetchRoot = true) {
     return async (dispatch, getState) => {
         const userId = getCurrentUserId(getState());
         let posts;
         try {
-            posts = await Client4.getPostsUnread(channelId, userId);
+            posts = await Client4.getPostsUnread(channelId, userId, 30, 30, fetchRoot);
```
Comment (jespino) [inline src/actions/posts.js]: Can we change the 30, 30 with constants?

File: package-lock.json (hunk 2 above)
Comment (reflog) [reply package-lock.json]: 🤦‍♂️

File: src/actions/posts.js (hunk 3 above)
Comment (reflog) [reply src/actions/posts.js]: will do

File: src/actions/posts.js [hunk 4]
```diff
@@ -601,12 +601,12 @@ export function getPostThread(rootId) {
     };
 }
 
-export function getPosts(channelId, page = 0, perPage = Posts.POST_CHUNK_SIZE) {
+export function getPosts(channelId, page = 0, perPage = Posts.POST_CHUNK_SIZE, fetchRoot = true) {
```
Comment (lieut-data) [inline src/actions/posts.js]: `fetchRoot` seems like a misnomer to me: should this parameter be called, `fetchThreads`?
